students/kmsnyde/lesson3/strformat_lab.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sun Apr  8 08:10:17 2018
+
+@author: Karl M. Snyder
+"""
+#[[fill]align][sign][#][0][minimumwidth][.precision][type]
+# use * to unpack tuple
+
+# TASK 1
+ 
+a_tuple = (2, 123.4567, 10000, 12345.67)
+print('file_{:03}: {:.2f}, {:.2e}, {:.2e}'.format(*a_tuple))
+ 
+# TASK 2
+
+#by name
+print('file_{num_file:03}: {_2_dec:.2f}, {_2_dec_e:.02e}, {_4digit_e:.3e}'.\
+      format(num_file=2, _2_dec=123.4567, _2_dec_e=10000, _4digit_e=12345.67))
+#by position
+print('file_{3:03}: {0:.2f}, {1:.2e}, {1:.3e}'.format(\
+      123.4567, 12345.67, 10000, 2))
+#f-strings
+num_file = 2
+_2_dec = 123.4567
+_2_dec_e = 10000
+_4digit_e = 123456.67
+print(f'file_{num_file:03}, {_2_dec:.2f}, {_2_dec_e:.02e}, {_4digit_e:.3e}')
```

Comment:
you could also access by index directly `f'file_{a_tuple[0]}'
or assign vars in bulk: 
`num_file, _2_dec, _2_dec_e, _4digit_e = a_tuple`